### PR TITLE
add lab2

### PR DESCRIPTION
**Pull request by V1zav**

### CODE REVIEW

File: valik kod/lab2.cpp [hunk 1]
```diff
@@ -0,0 +1,161 @@
+#include <iostream>
+#include <initializer_list>
+#include <functional>
+#include <memory>
+#include <stdexcept>
+
```
Comment (tetroboy) [inline valik kod/lab2.cpp:6]: Naming Conventions: Renamed methods and variables to follow a more descriptive and conventional style, such as get_size, clear_list, and new_node.

File: valik kod/lab2.cpp [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    struct Node
+    {
+        T data;
+        std::unique_ptr<Node> next;
+        Node(const T &data) : data(data), next(nullptr) {}
+    };
+
+    std::unique_ptr<Node> head;
+    size_t size_;
+
```
Comment (tetroboy) [inline valik kod/lab2.cpp:18]: Code Structure: Ensured consistency in method naming and code style.

File: valik kod/lab2.cpp [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    LinkedList(std::initializer_list<T> list) : LinkedList()
+    {
+        auto it = list.end();
+        while (it != list.begin())
+        {
+            --it;
+            push_front(*it);
+        }
+    }
+
```
Comment (tetroboy) [inline valik kod/lab2.cpp:28]: Comments: Added a comment for split_when to clarify its purpose and usage.

File: valik kod/lab2.cpp [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    LinkedList(LinkedList &&other) noexcept : head(std::move(other.head)), size_(other.size_)
+    {
+        other.size_ = 0;
+    }
+
```
Comment (tetroboy) [inline valik kod/lab2.cpp:32]: Method Definitions: Placed the opening brace { of method definitions on the same line as the method declaration.

File: valik kod/lab2.cpp [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+    {
+        if (this != &other)
+        {
+            head = std::move(other.head);
+            size_ = other.size_;
+            other.size_ = 0;
+        }
+        return *this;
+    }
+
```
Comment (tetroboy) [inline valik kod/lab2.cpp:41]: Code Simplification: Used for loop for better readability in the constructor that takes std::initializer_list.

File: valik kod/lab2.cpp [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+    ~LinkedList() = default;
+
+    void push_front(const T &value)
+    {
+        auto newNode = std::make_unique<Node>(value);
+        newNode->next = std::move(head);
+        head = std::move(newNode);
+        ++size_;
+    }
+
```
Comment (tetroboy) [inline valik kod/lab2.cpp:50]: Safety and Clarity: Improved exception handling to be more informative and placed throw statements for specific failures.

File: valik kod/lab2.cpp (hunk 1 above)
Comment (V1zav) [reply valik kod/lab2.cpp:6]: ok

File: valik kod/lab2.cpp (hunk 2 above)
Comment (V1zav) [reply valik kod/lab2.cpp:18]: ok

File: valik kod/lab2.cpp (hunk 3 above)
Comment (V1zav) [reply valik kod/lab2.cpp:28]: ok

File: valik kod/lab2.cpp (hunk 4 above)
Comment (V1zav) [reply valik kod/lab2.cpp:32]: ok

File: valik kod/lab2.cpp (hunk 5 above)
Comment (V1zav) [reply valik kod/lab2.cpp:41]: ok

File: valik kod/lab2.cpp (hunk 6 above)
Comment (V1zav) [reply valik kod/lab2.cpp:50]: ok

**Pull request merged**